Bit Manipulation/Single Number II.java

```diff
@@ -1,21 +1,14 @@
 public class Solution {
     // DO NOT MODIFY THE LIST. IT IS READ ONLY
     public static int singleNumber(final List<Integer> A) {
-        Map<Integer, Integer> map = new HashMap<>();
-
-        for (int num : A) {
-            map.put(num, map.getOrDefault(num, 0) + 1);
-        }
-
-        int ans = -1;
-
-        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
-            if (entry.getValue() == 1) {
-                ans = entry.getKey();
-                break;
+        int sum = 0;
+        for(int i=0;i<32;i++){
```

Comment:
Can you please add a comment about why this approach works?

Reply:
Okay